### DISCUSSION TITLE
luxon 3.3.0 method definitions vs types response mismatch

### DISCUSSION THREAD

**Discussion by davingee** [DefinitelyTyped/DefinitelyTyped#65947] [Issues with a @types package]
toISODate returns a string by toISODate types says string or null
```
toISODate(opts: Object): string
https://moment.github.io/luxon/api-docs/index.html#datetimetoisodate
@types
toISODate(opts?: ToISODateOptions): string | IfInvalid<null>
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e44c0db96c1074b5f5541f2c5a2e4f6039b71c23/types/luxon/src/datetime.d.ts#L1267
```

[a comment by an automated account is not shown]

**Comment by davingee** [comment]
the same issue with 
https://www.jsdocs.io/package/@types/luxon#DateTime.toISO
https://www.jsdocs.io/package/@types/luxon#DateTime.toISOTime
https://www.jsdocs.io/package/@types/luxon#DateTime.toISODate

**Comment by ghost** [comment]
See https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/65161#discussioncomment-5634371 to understand why it is typed to return `null`. 

`IfInvalid<null>` is a return type because if the `DateTime` object is invalid then these methods return `null`.
